ql/src/test/org/apache/hadoop/hive/ql/txn/compactor/CompactorTest.java

```diff
@@ -102,6 +102,11 @@ protected CompactorTest() throws Exception {
     tmpdir.deleteOnExit();
   }
 
+  protected void deleteTableDir(String tableName) throws IOException {
```

Comment:
I agree with temp directory creation using Files.createTempDirectory. With this, we can ensure both concurrent and repeated calls won't end up with same directory. Also, will delete the temp directory as part of tearDown itself to avoid creating too many directories.